README.md

```diff
@@ -59,6 +59,13 @@ compile with 1.29.0 you will need to run the following version-pinning command:
 cargo update -p cc --precise "1.0.41" --verbose
 ```
 
+In order to use the `use-serde` feature or to build the unit tests with 1.29.0,
+the following version-pinning commands are also needed:
+```
+cargo update --package "serde" --precise "1.0.98"
```

Comment:
I don't think there's any reason to downgrade serde. I am testing on 1.29 fine without it.

Reply:
If I drop that line, I get this:
```
error: no matching version `= 1.0.118` found for package `serde_derive`
location searched: registry `https://github.com/rust-lang/crates.io-index`
versions found: 1.0.98
required by package `serde v1.0.118`
    ... which is depended on by `bitcoin v0.25.2 (file:///home/steven/code/rust/bitcoin)`
```

Because a higher serde will require a higher serde_derive. Somehow Cargo is not very smart about all that and it can't realize that it has a pin on _derive and then look for a matching serde.

Reply:
The script I use is
```
set -e

MSRV="1.29.0"

CMD="rustup run ${MSRV}"

rm -f Cargo.lock
$CMD cargo generate-lockfile
$CMD cargo update --package "cc" --precise "1.0.41"
$CMD cargo update --package "serde" --precise "1.0.98"
$CMD cargo update --package "serde_derive" --precise "1.0.98"

$CMD cargo test --features "use-serde rand base64"
```

Perhaps we should commit that.

Reply:
None of my scripts pin `serde` and everything works fine for me.

Reply:
Oh, I'm lying, it's just that my scripts never made any progress on this PR because of the other compilation issue.

Reply:
I just tried and this worked:
`rm -rf target/ && rm Cargo.lock && cargo +1.29.0 test`

Reply:
@elichai you have to add `--features=use-serde` to see the failure